### PR TITLE
fix: render markdown in og images

### LINKED ISSUE #7057: [CAL-1044] remove markdown from description and convert to html for social preview
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1044](https://linear.app/calcom/issue/CAL-1044/remove-markdown-from-description-and-convert-to-html-for-social)</sub>

### LINKED ISSUE #7717: [CAL-1245] meta tag content for social cards not parsed properly
### Issue Summary

I found a bug where setting part of your description to be bold isn't parsed properly in the thumbnail generation

image

Reported on slack:- https://calendso.slack.com/archives/C01NXDSRY9W/p1678760616978939

<sub>[CAL-1245](https://linear.app/calcom/issue/CAL-1245/meta-tag-content-for-social-cards-not-parsed-properly)</sub>

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/pull/7693

this issue was closed by #7474  but it's not fixed, this PR aim's to fix that. also closes #7057 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/pages/api/social/og/image.tsx [hunk 1]
```diff
@@ -108,7 +107,7 @@ export default async function handler(req: NextApiRequest) {
         description: searchParams.get("description"),
         imageType,
       });
-      const description_ = await (await remark().use(strip).process(description)).toString();
+      const description_ = md.render(description).replace(/(<([^>]+)>)/gi, "");
```
Comment (G3root) [inline apps/web/pages/api/social/og/image.tsx:110]: regex from https://github.com/lodash/lodash/issues/1361#issue-96609415

**Comment by G3root** [comment]
to test go to this url locally http://localhost:3000/api/social/og/image?type=meeting&title=**super**%20%20%3Cp%3E%3Cbr%3E%20%3C/p%3E%20%20long%20event%20title%20for%20testing%20purposes&meetingProfileName=Pro%20Example&meetingImage=http://localhost:3000/pro/avatar.png&names=Pro%20Example&usernames=pro

File: apps/web/pages/api/social/og/image.tsx [hunk 2]
```diff
@@ -75,10 +74,13 @@ export default async function handler(req: NextApiRequest) {
         meetingImage: searchParams.get("meetingImage"),
         imageType,
       });
+
+      const title_ = md.render(title).replace(/(<([^>]+)>)/gi, "");
```
Comment (G3root) [inline apps/web/pages/api/social/og/image.tsx:78]: regex from https://github.com/lodash/lodash/issues/1361#issue-96609415

**Review by CarinaWolli — COMMENTED** [review]
Tested the link and looks good 👍

@zomars are we updating and committing the yarn.lock file when we have changes in package.json?

**Comment by PeerRich** [comment]
> Tested the link and looks good 👍
> 
> @zomars are we updating and committing the yarn.lock file when we have changes in package.json?

I think we have to commit yarn.lock if package.json changes, yes

**Review by CarinaWolli — APPROVED** [review]

File: apps/web/pages/api/social/og/image.tsx [hunk 3]
```diff
@@ -108,8 +110,8 @@ export default async function handler(req: NextApiRequest) {
         description: searchParams.get("description"),
         imageType,
       });
-      const description_ = await (await remark().use(strip).process(description)).toString();
-      const img = new ImageResponse(<Generic title={title} description={description_} />, ogConfig) as {
+
+      const img = new ImageResponse(<Generic title={title} description={description} />, ogConfig) as {
```
Comment (zomars) [inline apps/web/pages/api/social/og/image.tsx:114]: Is this expected?

Comment (G3root) [reply apps/web/pages/api/social/og/image.tsx:114]: It is expected.. On my last PR to fix this problem I was cleaning up on the wrong template 😫, so I reverted back to how it was .

**Review by zomars — APPROVED** [review]
LGTM

**Comment by zomars** [comment]
Reverter yarn changes since it was causing type errors for some reason

**Pull request merged**